subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -157,6 +157,28 @@ Now that the build logic has been translated into a plugin, you can apply it in
 </sample>
 ++++
 
+[[sec:avoiding_use_of_gradlebuild]]
+=== Avoiding the use of `GradleBuild`
+
+A build script can define task for invocation other Gradle builds through the task type api:org.gradle.api.tasks.GradleBuild[]. The use of this type is generally discouraged. There are some corner cases where the invoked build doesn't expose the same runtime behavior as from the command line or through the Tooling API leading to unexpected results.
+
+Usually, there's a better way to model the requirement. The appropriate approach depends on the problem at hand. Here're some options:
+
+* Model the build as <<multi_project_builds,multi-project build>> if the intention is to execute tasks from different modules as unified build.
+* Use <<composite_builds,composite builds>> for projects that are physically separated but should occasionally be built as a single unit.
+
+[[sec:avoiding_inter_project_configuration]]
+=== Avoiding inter-project configuration
+
+Gradle does not restrict build script author to reach into the domain model from one project into another one in a <<multi_project_builds,multi-project build>>. Strongly-coupled projects hurts <<sec:parallel_execution,build execution performance>> as well as readability and maintainability of code.
+
+The following practices should be avoided:
+
+* Explicitly depending on a task from another project via api:org.gradle.api.Task#dependsOn(java.lang.Object...)[].
+* Setting property values or calling methods on domain objects from another project.
+* Executing another portion of the build with <<sec:avoiding_use_of_gradlebuild,GradleBuild>>.
+* Declaring unnecessary <<sec:declaring_project_dependency,project dependencies>>.
```

Comment:
❓ Shall we also mention `evaluationDependsOn()` and `evaluationDependsOnChildren()` here?

Reply:
That's likely a good idea though it would definitely also require to change the [chapter on multi-project builds](https://docs.gradle.org/current/userguide/multi_project_builds.html#sub:configuration_time_dependencies) which covers both methods in detail.

What's our recommendation to use instead? Should we deprecate those methods?

Reply:
I would love we deprecate and remove those.
But in the meantime I think they should be mentioned here.

Reply:
If we do mention them here then I'd like to propose how to solve the specific problem with a recommended solution. What do we tell build script authors to do instead?

Reply:
It's about the same as the first bullet point of this list. It is about coupling project configuration phases. If you require `evaluationDependsOn()` it is probably because you need to model something better in your build, missing project? publications? etc...

What it does is imperatively pause current project evaluation and force evaluation of the depended on projects. And you can use it about anywhere.

My only concern is that mentioning those methods is considered. This should not block this PR.

Reply:
> it is probably because you need to model something better in your build, missing project? publications?

We'll definitely need to document what you _should_ do instead. Just saying that you shouldn't use a public, non-deprecated API won't cut it. I think it would rather lead to more confusion especially because we explain the exact feature in a different chapter in detail. I'd rather address it as a separate PR that explains all of the use cases and provides a holistic solution.

Reply:
SGTM